fhir-server/src/main/java/org/linuxforhealth/fhir/server/util/RestAuditLogger.java

```diff
@@ -746,6 +746,64 @@ protected static AuditLogEntry populateAuditLogEntry(AuditLogEntry entry, HttpSe
         return entry;
     }
 
+    /**
+     * Populates the passed audit log entry for Bundle entry.
+     * This method will populate the resource id, resource type and version id from the
+     * Bundle response entry location when the return preference is "OperationOutcome".
+     * When the return preference is "representation" the populateAuditLogEntry method will
+     * populate the resource id, resource type and version id.
+     * @param entry
+     *  The AuditLogEntry to be populated.
+     * @param responseEntry
+     *  The Bundle response entry.
+     * @param request
+     *  The HttpServletRequest representation of the REST request.
+     * @param resource
+     *  The Resource object.
+     * @param startTime
+     *  The start time of the request execution.
+     * @param endTime
+     *  The end time of the request execution.
+     * @param responseStatus
+     *  The response status.
+     * @return AuditLogEntry - an initialized audit log entry.
+     */
+    protected static AuditLogEntry populateBundleAuditLogEntry(AuditLogEntry entry, Bundle.Entry responseEntry, HttpServletRequest request, Resource resource,
+        Date startTime, Date endTime, Response.Status responseStatus) { 
+        final String METHODNAME = "populateBundleAuditLogEntry";
+        log.entering(CLASSNAME, METHODNAME);
+        
+        // call populateAuditLogEntry to populate common attributes to all rest
+        populateAuditLogEntry(entry, request, resource, startTime, endTime, responseStatus);
+        
+        // Populate the resource id, resource type and version id from the
+        // Bundle response entry location when the return preference is "OperationOutcome".
+        if (HTTPReturnPreference.OPERATION_OUTCOME.equals(FHIRRequestContext.get().getReturnPreference())) {
+            if (responseEntry.getResponse() != null && responseEntry.getResponse().getLocation() != null) {
+                String location = responseEntry.getResponse().getLocation().getValue();
+                String[] parts = location.split("/");
+                String resourceType = null;
+                String id = null;
+                String versionId = null;
+                if (parts.length == 10) {
+                    resourceType = parts[6];
+                    id = parts[7];
+                    versionId = parts[9];
+                } else if (parts.length == 4) {
+                    resourceType = parts[0];
+                    id = parts[1];
+                    versionId = parts[3];
+                }
```

Comment:
we try to avoid logic this that assumes a particular path for the base url (I think technically user could change the context path or have some rewrite rules that complicate things)

Reply:
This has been fixed. As discussed, the logic to read resource id, type and version id has been changed to read from the reverse of location string.